Framework/PythonInterface/plugins/algorithms/CalculateEfficiencyCorrection.py

```diff
@@ -0,0 +1,218 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import absolute_import, division, print_function
+import numpy as np
+
+from mantid.simpleapi import \
+    CloneWorkspace, ConvertFromDistribution, ConvertToPointData, ConvertUnits, SetSampleMaterial
+from mantid.api import \
+    mtd, AlgorithmFactory, DataProcessorAlgorithm, MatrixWorkspaceProperty
+from mantid.kernel import \
+    Direction, FloatBoundedValidator, MaterialBuilder, StringMandatoryValidator
+
+PROPS_FOR_TRANS = ['DensityType']
+TABULATED_WAVELENGTH = 1.7982
+
+
+class CalculateEfficiencyCorrection(DataProcessorAlgorithm):
```

Comment:
I think the reason I did this was to use the `copyProperties` [here](https://github.com/mantidproject/mantid/blob/b09e102f5b2180d4002f9c51cb816491e9affe54/Framework/PythonInterface/plugins/algorithms/CalculateEfficiencyCorrection.py#L91) but initially had more properties copied over. Probably a bad idea since depends on another algorithm that could be deprecated. 

I will just define property and switch to `PythonAlgorithm`, thanks.

Reply:
Being a `DataProcessorAlgorithm` also affects how the history is recorder: you can unroll the history for such algorithms and see all the calls to child algorithms as well. I don't think this functionality is needed here.